tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -813,6 +813,116 @@ function test30() {
     assert(y, 117);
 }
 
+type School record {|
+    string name;
+|};
+
+type Doctor record {
+    string name;
+    string category;
+    School school;
+};
+
+function testClosureWithStructuredBindingTypeParams() {
```

Comment:
TypeParams mean something else in b7a, so may be confusing. Also it is not the param itself that is the reference?

Reply:
Updated to `testClosureWithStructuredBindingVars`